app/src/main/java/com/example/habitsmasher/MainActivity.java

```diff
@@ -86,10 +89,40 @@ public boolean onOptionsItemSelected(MenuItem item) {
             openFollowUserDialog();
             return true;
         }
+        if (item.getItemId() == R.id.logout_button) {
+            FirebaseAuth.getInstance().signOut();
+            _navController.navigate(R.id.action_logout);
+        }
 
         return super.onOptionsItemSelected(item);
     }
 
+    /**
+     * This method is called when the back button is pressed in the application
+     */
+    @Override
+    public void onBackPressed(){
+        // if the back button is pressed on any top level destinations, do nothing
+        if (!isControllerOnTopLevelDestination()) {
+            super.onBackPressed();
+        }
+    }
+
+    /**
+     * This method checks if the application is on any of the specified top-level nav destinations
+     * @return boolean true if on top-level, false otherwise
+     */
+    private boolean isControllerOnTopLevelDestination() {
+        return Objects.requireNonNull(_navController.getCurrentDestination())
+                      .getId() == R.id.user_login ||
+                Objects.requireNonNull(_navController.getCurrentDestination())
+                       .getId() == R.id.navigation_dashboard ||
+                Objects.requireNonNull(_navController.getCurrentDestination())
+                       .getId() == R.id.navigation_home ||
+                Objects.requireNonNull(_navController.getCurrentDestination())
+                       .getId() == R.id.navigation_notifications;
+    }
```

Comment:
This chunk of code ensures that the **back button** is disabled on any "top-level destinations". You shouldn't be able to hit the "back button" after you log out to automatically jump back into the app.